Framework/Algorithms/src/ResampleX.cpp

```diff
@@ -114,8 +114,21 @@ map<string, string> ResampleX::validateInputs() {
  */
 string determineXMinMax(MatrixWorkspace_sptr inputWS, vector<double> &xmins,
                         vector<double> &xmaxs) {
-  bool updateXMins = xmins.empty(); // they weren't set
-  bool updateXMaxs = xmaxs.empty(); // they weren't set
+  const size_t numSpectra = inputWS->getNumberHistograms();
+
+  // pad out the ranges by copying the first value to the rest that are needed
+  if (xmins.size() == 1 && numSpectra > xmins.size()) {
+    const double value = xmins.front();
+    xmins.insert(xmins.end(), numSpectra - xmins.size(), value);
+  }
+  if (xmaxs.size() == 1 && numSpectra > xmaxs.size()) {
+    const double value = xmaxs.front();
+    xmaxs.insert(xmaxs.end(), numSpectra - xmaxs.size(), value);
+  }
+
```

Comment:
Please update the documentation for each parameter.

Reply:
The [algorithm documentation](http://docs.mantidproject.org/nightly/algorithms/ResampleX-v1.html) already states this. It was missing behavior that was only recently noticed.

Reply:
The fact is was only recently noticed suggests we need to make it more visible in the documentation, such as in the parameter description. At the same time I worry about getting too deeply into the realm of yak shaving.